lib/puppet/transaction/report.rb

```diff
@@ -158,6 +168,21 @@ def compute_status(resource_metrics, change_metric)
   end
 
   # @api private
+  def has_noop_change_events?(resource)
+    resource.events.any? { |event| (event.status == 'noop' && event.name == :content_changed) }
```

Comment:
Events that include changes don't always have a `name` field with value `:content_changed`, in fact, that field is a composition of `property + "_changed"` (a little useless I know). So actually, you're only catching `content` field change events specifically with this mechanism only, I presume you were testing with `file { "foo": content => "blah" }`.